Assets/MRTK/Core/Providers/BaseGenericInputSource.cs

```diff
@@ -20,7 +20,18 @@ public BaseGenericInputSource(string name, IMixedRealityPointer[] pointers = nul
         {
             SourceId = (CoreServices.InputSystem != null) ? CoreServices.InputSystem.GenerateNewSourceId() : 0;
             SourceName = name;
-            Pointers = pointers ?? new[] { CoreServices.InputSystem?.GazeProvider?.GazePointer };
+            if (pointers != null)
+            {
+                Pointers = pointers;
+            }
+            else if (CoreServices.InputSystem?.GazeProvider?.GazePointer is IMixedRealityPointer gazePointer)
```

Comment:
Is this a valid state? Do we ever want the gaze provider or gaze pointer to _not_ be present?

Reply:
I think there might be a scenario in the future where the gaze provider _could_ be a toggleable feature. This block of code was originally here to safeguard against the GazeProvider not getting instantiated due to the type not being set. Though we've addressed this case in the editor, I still think that it's smart to safeguard against it here as well in case of other unexplored scenarios

Reply:
>I still think that it's smart to safeguard against it here as well in case of other unexplored scenarios

I think that's the root of my uncertainty: if this is an unexplored scenario, should we even attempt to support it? Or should it be more of a "something went wrong; the gaze provider/pointer is required"?

Reply:
That's a good point, though as far as scenarios where we might not need/want the gaze provider/gaze pointer, VR applications (some WMR devices, Oculus) don't make use of the gaze pointer/provider, and I think as we move to become more modular it might make sense to disable it when deploying to those devices. We'd probably need to rework the focus provider a bit more to not target the gaze provider so much, but this would at least be a first step towards a refactor of those components.